turbinia/workers/analysis/loki.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task for running Loki on drives & directories."""
+
+import csv
+import os
+from turbinia import TurbiniaException
+
+from turbinia.evidence import EvidenceState as state
+from turbinia.evidence import ReportText
+from turbinia.lib import text_formatter as fmt
+from turbinia.workers import Priority
+from turbinia.workers import TurbiniaTask
+
+
+class LokiAnalysisTask(TurbiniaTask):
+  """Task to use Loki to analyse files."""
+
+  REQUIRED_STATES = [
+      state.ATTACHED, state.MOUNTED, state.CONTAINER_MOUNTED, state.DECOMPRESSED
+  ]
+
+  def run(self, evidence, result):
+    """Run the Loki worker.
+
+    Args:
+        evidence (Evidence object):  The evidence to process
+        result (TurbiniaTaskResult): The object to place task results into.
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    # Where to store the resulting output file.
+    output_file_name = 'loki_analysis.txt'
+    output_file_path = os.path.join(self.output_dir, output_file_name)
+
+    # What type of evidence we should output.
+    output_evidence = ReportText(source_path=output_file_path)
+
+    self.updateLoki(result)
+
+    try:
+      (report, priority, summary) = self.runLoki(result, evidence)
+    except TurbiniaException as e:
+      result.close(
+          self, success=False, status='Unable to run Loki: {0:s}'.format(
+              str(e)))
+      return result
+
+    output_evidence.text_data = report
+    result.report_priority = priority
+    result.report_data = report
+
+    # Write the report to the output file.
+    with open(output_file_path, 'wb') as fh:
+      fh.write(output_evidence.text_data.encode('utf-8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True, status=summary)
+    return result
+
+  def updateLoki(self, result):
+    cmd = ['python', os.path.expanduser('~/Loki/loki-upgrader.py')]
+    self.execute(cmd, result, cwd=os.path.expanduser('~/Loki/'))
+
+  def runLoki(self, result, evidence):
+    log_file = os.path.join(self.output_dir, 'loki.log')
+    stdout_file = os.path.join(self.output_dir, 'loki_stdout.log')
+    stderr_file = os.path.join(self.output_dir, 'loki_stderr.log')
+
+    cmd = [
+        'python',
+        os.path.expanduser('~/Loki/loki.py'), '-w', '0', '--csv', '--intense',
```

Comment:
Is there a way we can make this (and the upgrader script) more flexible by maybe installing the package in a more standard way so we don't have to hard-code the paths?

Reply:
Can you also add the required binaries to the `DEPENDENCIES` in `turbinia_config_tmpl.py`?

Reply:
I could unzip it to something like /usr/bin so its on the PATH?